src/codegen/function_body_chunk.rs

```diff
@@ -159,6 +159,24 @@ impl Builder {
                     }
                 }
             }).collect();
+
+        if trampoline.use_ret {
+            if let Some(ref return_value) = trampoline.return_value {
```

Comment:
Hm, there way to have use_ret=true and return_value = None ?
Maybe we can combine this fields to one return_value?

Reply:
the opposite could be true yes (return_value that shouldn't be used), I could pass only the return_value when it needs to be used, but then that would make things a bit unintuitive.

Reply:
As I know it normal in languages with optional types that if `return_value = Some(parameter)` then it used and `return_value = None` meant that nothing to return, so separate boolean field on the contrary makes it more confusing

Reply:
in that context, return_value is the return_value of the finish function, not something we have to "return", that's consistent with the other fields in the trampoline object (`output_params` in particular)

Reply:
Yes, from trampoline point of view it parameter for result for ffi-function (xxx_finish) and from other side it main type for good case. So IMHO we can use `ffi_ret` as field name, as I don't see normal way to separate parts of `Ok` content

Reply:
I don't understand what you're saying here :)

Reply:
I says that this problem field in async trampoline (currently named `return_value`) is parameter that returns from FFI-function (xx_finish) in your case, so we can name it `ffi_ret` or `finish_ret` and use it in same way as now but without `use_ret` flag

Reply:
.. but set `ffi_ret` to `Some` only if `use_function_return_for_result` return true

Reply:
Well if you insist on that I don't really mind, still don't like it though ;)